evalml/pipelines/components/transformers/preprocessing/featuretools.py

```diff
@@ -72,9 +94,18 @@ def transform(self, X, y=None):
         """
         X_ww = infer_feature_types(X)
         X_ww = X_ww.ww.rename({col: str(col) for col in X_ww.columns})
-        es = self._make_entity_set(X_ww)
 
-        feature_matrix = calculate_feature_matrix(features=self.features, entityset=es)
+        features_to_use = (
```

Comment:
Not sure about this, but would it make sense to do the filtering during `fit` instead of here, so that  after calling `fit`, `self.features` would always contain the features that were actually used by the transformer? You could change `self._passed_in_features` to retain the original list of features that were provided if you need to hold on to those, and if the list is present or contains values you know the user passed in features, otherwise you know they were generated from DFS.

Reply:
@thehomebrewnerd thought about this a little more and decided to keep the filtering on the transform end. Main reasoning being that ultimately we need to look at the columns that are passed into `transform` to decide on which features to use. I could add logic verifying that the columns between the dataframe for `fit` and `transform` are the same but I'd rather keep it simple and keep `fit` a no-op and do the filtering in `transform`. Let me know if that makes sense or if you other suggestions!

Reply:
Ah, I guess I wasn't thinking about the set of columns used for `fit` being different from the set of columns passed to `transform`.  I don't have any strong feelings one way or the other here, so I'm fine if you keep the filtering in `transform`.